Source/Libraries/FaultData/DataResources/EventClassificationResource.cs

```diff
@@ -165,31 +165,18 @@ private EventClassification Classify(MeterDataSet meterDataSet, DataGroup dataGr
 
         private static bool ValidateEventType(EventClassification eventClassification, Asset asset)
         {
-            AssetType assetType = (AssetType)asset.AssetTypeID;
+            if (eventClassification == EventClassification.Other)
+                return true;
 
-            switch (eventClassification)
+            using (AdoDataConnection connection = asset.ConnectionFactory())
             {
-                case EventClassification.BreakerOpen:
-                    return assetType == AssetType.Breaker;
+                EventType evtType = new TableOperations<EventType>(connection).QueryRecordWhere("Name = {0}", eventClassification.ToString());
```

Comment:
`EventOperation.GetEvents()` does call `TableOperations<EventType>.GetOrAdd()`, but `ValidateEventType()` is called by `EventClassificationResource.Initialize()` which happens before that. Therefore, we have to assume that `evtType` could be null as well.